src/Core/Content/Seo/SeoUrlGenerator.php

```diff
@@ -26,31 +29,40 @@ class SeoUrlGenerator
 
     private RouterInterface $router;
 
-    private Environment $twig;
-
     private DefinitionInstanceRegistry $definitionRegistry;
 
     private RequestStack $requestStack;
 
+    private SeoUrlTwigFactory $seoUrlTwigFactory;
+
+    private SlugifyInterface $slugify;
+
+    private iterable $twigExtensions;
+
+    private LoggerInterface $logger;
+
     public function __construct(
         DefinitionInstanceRegistry $definitionRegistry,
         RouterInterface $router,
         RequestStack $requestStack,
-        Environment $environment
+        SeoUrlTwigFactory $seoUrlTwigFactory,
+        SlugifyInterface $slugify,
+        iterable $twigExtensions,
+        LoggerInterface $logger
     ) {
         $this->definitionRegistry = $definitionRegistry;
         $this->router = $router;
         $this->requestStack = $requestStack;
-        $this->twig = $environment;
+        $this->seoUrlTwigFactory = $seoUrlTwigFactory;
+        $this->slugify = $slugify;
+        $this->twigExtensions = $twigExtensions;
+        $this->logger = $logger;
     }
 
-    /**
-     * @feature-deprecated (flag:FEATURE_NEXT_13410) Parameter $salesChannel will be required
-     */
     public function generate(array $ids, string $template, SeoUrlRouteInterface $route, Context $context, ?SalesChannelEntity $salesChannel): iterable
     {
         $criteria = new Criteria($ids);
-        $route->prepareCriteria($criteria, $salesChannel);
```

Comment:
revert this change

Reply:
Yes, sorry. That is from my local shopware 6.3 fix.